### PR TITLE
Remove "window.rollexIntervals"

### LINKED ISSUE #19: Use singleton class instead of global objects
Currently we use `window.rollexIntervals` object for making multiple counters run on a single interval. It would be better to rewrite this with a singleton class to avoid `window` pollution.

### PR DESCRIPTION
**Pull request by akxcv**
Fixes #19

### CODE REVIEW

**Comment by akxcv** [comment]
The performance on low-end mobile devices is terrible. Tested on iPhone 4S with just 11 counters on a single page (5 static, 5 animated, 1 frozen). We'll need to find what's causing this.

`tick()` might perform too slow, and `setInterval` just keeps calling it again and again, even if it did not execute yet. This might completely freeze the tab on low-end devices.

Also, the instant zero reset hack is not working on iPhone 4. Did not test it on a single counter though, so it might just be because of overall poor performance.

File: test/unit/counter/ticking.test.js [hunk 1]
```diff
@@ -118,4 +85,49 @@ describe('counting', function () {
...
+    it('does not count when "frozen" is true', function () {
+      component = mount(<Counter seconds={10} frozen />)
+      jest.runTimersToTime(5000)
+      expect(component.state().timeDiff).toBe(10000)
+      expect(component.state().numbers.seconds).toBe(10)
+    })
+
+    it('stops to count if frozen prop is updated to true', function () {
+      component = mount(<Counter from={0} to={10000} />)
+      component.instance().stop = jest.genMockFunction()
```
Comment (JelF) [inline test/unit/counter/ticking.test.js]: What's the reason?

**Comment by JelF** [comment]
> The performance on low-end mobile devices is terrible. Tested on iPhone 4S with just 11 counters on a single page (5 static, 5 animated, 1 frozen). We'll need to find what's causing this.

have you checked it with different easing functions?

File: test/unit/counter/ticking.test.js (hunk 1 above)
Comment (akxcv) [inline test/unit/counter/ticking.test.js]: The easiest way to ensure that all the components are unmounted after each test is to mount components into the same variable every time. Then we can just do

```js
afterEach(function () {
  component.unmount()
})
```

**Comment by akxcv** [comment]
> have you checked it with different easing functions?

Yes. But it seems that it's the easing *duration*, not *function*, that matters the most. If easing duration is bigger, the timer is more prone to glitching.

File: test/unit/counter/ticking.test.js (hunk 1 above)
Comment (JelF) [inline test/unit/counter/ticking.test.js]: i mean, what's the reason of mocking stop?

Comment (JelF) [inline test/unit/counter/ticking.test.js]: Sorry, i missed the check for calls done

**Comment by akxcv** [comment]
The performance problem described above is a separate issue. I haven't yet managed to find the source of it. We'll need some more elaborate testing to fix it.

**Pull request merged**